tests/rkt_volume_mount_test.go

```diff
@@ -0,0 +1,331 @@
+			},
+			"",
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{
+						Name: baseAppName,
+						App: &types.App{
+							Exec:  []string{"/inspect", "--write-file", "--read-file"},
+							User:  "0",
+							Group: "0",
+							Environment: []types.EnvironmentVariable{
+								{"FILE", path.Join(tmpdirpathpod, "file")},
+								{"CONTENT", "host:foo"},
+							},
+							MountPoints: []types.MountPoint{
+								{tmpdirmountname, tmpdirpathpod, false},
+							},
+						},
+						ReadOnlyRootFS: true,
+					},
+				},
+				Volumes: []types.Volume{
+					{tmpdirmountname, "host", tmpdir, nil, nil, nil, nil, nil},
+				},
+			},
+			0,
+			"host:foo",
+		},
+	}
+
+	volumeMountTestCasesNonRecursive = []volumeMountTestCase{
+		{
+			"Read of nested file for non-recursive mount",
+			[]imagePatch{
+				{"rkt-test-run-pod-manifest-recursive-mount-stat.aci", []string{}},
+			},
+			"",
+			&schema.PodManifest{
+				Apps: []schema.RuntimeApp{
+					{
+						Name: baseAppName,
+						App: &types.App{
+							Exec:  []string{"/inspect", "--read-file"},
+							User:  "0",
+							Group: "0",
+							Environment: []types.EnvironmentVariable{
+								{"FILE", tmpdir2filepathpod},
+							},
+							MountPoints: []types.MountPoint{
+								{"dir1", tmpdirpathpod, false},
+							},
+						},
+					},
+				},
+				Volumes: []types.Volume{
+					{tmpdirmountname, "host", tmpdir, nil, &boolFalse, nil, nil, nil},
+				},
+			},
+			0,
+			tmpdir2outerfilecontent,
+		},
+	}
+)
+
+func NewTestVolumeMount(volumeMountTestCases [][]volumeMountTestCase) testutils.Test {
+	return testutils.TestFunc(func(t *testing.T) {
+		ctx := testutils.NewRktRunCtx()
+		defer ctx.Cleanup()
+
+		deferredFuncs := prepareTmpDirWithRecursiveMountsAndFiles(t)
+		defer executeFuncs(deferredFuncs)
+
+		for _, testCases := range volumeMountTestCases {
+			for i, tt := range testCases {
+				var hashesToRemove []string
+				for j, v := range tt.images {
+					hash := patchImportAndFetchHash(v.name, v.patches, t, ctx)
+					hashesToRemove = append(hashesToRemove, hash)
+					if tt.podManifest != nil {
+						imgName := types.MustACIdentifier(v.name)
+						imgID, err := types.NewHash(hash)
+						if err != nil {
+							t.Fatalf("Cannot generate types.Hash from %v: %v", hash, err)
+						}
+						tt.podManifest.Apps[j].Image.Name = imgName
+						tt.podManifest.Apps[j].Image.ID = *imgID
+					}
+				}
+
+				manifestFile := ""
+				if tt.podManifest != nil {
+					tt.podManifest.ACKind = schema.PodManifestKind
+					tt.podManifest.ACVersion = schema.AppContainerVersion
+
+					manifestFile = generatePodManifestFile(t, tt.podManifest)
+					defer os.Remove(manifestFile)
+				}
+
+				// 1. Test 'rkt run'.
+				runCmd := fmt.Sprintf("%s run --mds-register=false", ctx.Cmd())
+				if manifestFile != "" {
+					runCmd += fmt.Sprintf(" --pod-manifest=%s", manifestFile)
+				} else {
+					runCmd += fmt.Sprintf(" %s %s", tt.cmdArgs, hashesToRemove[0])
```

Comment:
`hash` is defined in the previous for-loop and not available here. This code doesn't actually run the test for each image provided, but only for the first one. This can be improved and I'll add a comment.